File: Java/BOJ/1246 온라인 판매.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int[] P = new int[M];
        for(int i = 0 ; i < M; i++) {
        	P[i] = Integer.parseInt(br.readLine());
        }
        Arrays.sort(P);
        int result = 0;
        int cost = 0;
        for (int i = 0 ; i < M; i++) {
        	int temp = M - i;
        	if (temp <= N && P[i] * temp > result) {
        		result = temp*P[i];
        		cost = P[i];
        	}
        }
        System.out.print(cost + " " + result);
    }
}

File: Python/BOJ/1246 온라인 판매.py
import sys
N, M = map(int, sys.stdin.readline().split())
P = sorted([int(sys.stdin.readline()) for _ in range(M)])
cost = 0
result = 0
for i in range(M):
    temp = M - i
    if N >= temp and P[i] * temp > result :
        cost = P[i]
        result = P[i] * temp
print(cost,result)